src/main/bash/sdkman-install.sh

```diff
@@ -77,26 +77,35 @@ function __sdkman_install_candidate_version {
 }
 
 function __sdkman_install_local_version {
-	local candidate version folder
+	local candidate version folder versionLength versionLengthMax
+
+	versionLengthMax=15
 
 	candidate="$1"
 	version="$2"
 	folder="$3"
 
-	mkdir -p "${SDKMAN_CANDIDATES_DIR}/${candidate}"
+	#Validate max length of version is 15
+	versionLength=${#version}
 
-	# handle relative paths
-	if [[ "$folder" != /* ]]; then
-		folder="$(pwd)/$folder"
-	fi
+	if [[ $versionLength > $versionLengthMax ]]; then
+		__sdkman_echo_red "Invalid version! ${version} with length ${versionLength} exceeds max of ${versionLengthMax}!"
+	else
```

Comment:
It might be worth escaping with a non-zero exit code here rather than relying on an else statement. Here is an example of how we've done it elsewhere: https://github.com/sdkman/sdkman-cli/blob/master/src/main/bash/sdkman-install.sh#L57